src/uproot/source/fsspec.py

```diff
@@ -37,12 +51,22 @@ def __repr__(self):
             path = repr("..." + self._file_path[-10:])
         return f"<{type(self).__name__} {path} at 0x{id(self):012x}>"
 
+    def __getstate__(self):
+        state = dict(self.__dict__)
+        state.pop("_executor")
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        self._open()
+
```

Comment:
```suggestion
```

I don't see any reason why a `Source` should be pickleable.

Reply:
I just implemented it because it was implemented in other sources. I guess I can safely remove those in this PR since it's not being use anywhere.

Reply:
We'll keep pickleability.